File: chapter02/main/src/main/java/game/Deck.java
package game;

import java.util.List;
import java.util.Random;

class Deck {

  private static Random random = new Random();

  private List<Integer> cards;

  Deck() {
    // init with cards
  }

  Card drawCard() {
    return new Card(random.nextInt(4), random.nextInt(13) + 1);
  }

}

File: chapter02/main/src/main/java/game/Deck.kt
package game

import java.util.*

internal class Deck {
    private var cards: MutableList<Card> = ArrayList()

    init {
        // init with cards
        for (i in 1..13) {
            for (j in 0..3) {
                this.cards.add(Card(j, i))
            }
        }
        cards.shuffle()
    }

    fun drawCard(): Card {
        return this.cards.removeAt(0)
    }
}
